### DISCUSSION TITLE
Elsa architectural understanding

### DISCUSSION THREAD

**Discussion by nebula2** [elsa-workflows/elsa-core#3606] [Q&A]
Got a question regarding elsa.
if it is better to place this as a discussion in github please let me know 🙂

I am currently developing an application where I need workflow capabilities. I have started with something self made using rabbitmq - but as requirements grow it's getting out of hand.
Wanted to use elsa for a while now and so I started to take a deeper look.

Seems like elsa can do everything I need - but I struggle to understand the architecture. Is it possible to seperate elsa server, elsa dashboard and elsa worker?
I attached an image which hopefully explains my desired outcome.

![image](https://user-images.githubusercontent.com/5841525/211110415-ff55f5df-3607-44c6-b1dd-197d3986d077.png)

- Elsa workers consumes and processes messages from a queue
- Elsa server holds workflow definitions and is responsible for spawning messages as well as giving other services an API to interact with

Aleady asked this question on discord and it seems to be possible - but I need some guidance/ direction on where to start as I struggle to find examples/ approaches.

Thank you for your time :)

**Comment by rattrick1** [comment]
I have a similar need in that I want to have two different systems that process completely different types of jobs (Workflows & Activities designed in completely separate projects), but want one dashboard to view & manage all workflows.

**Comment by sjd2021** [comment]
I asked the same question a bit ago and the solution I'm using is to create a proxy endpoint in my user-facing site which adds authentication header values and forwards it to the workflow server that does the processing and has all the workflow endpoints available. It's not an amazing solution, but it fits my needs.
